### PR TITLE
chore: migrate to foundry

### PR DESCRIPTION
**Pull request by sujithsomraaj**
- removed all hardhat scripts
- removed hardhat installation guide
- added openzeppelin module

### CODE REVIEW

File: src/MultiMessageReceiver.sol [hunk 1]
```diff
@@ -106,65 +120,50 @@ contract MultiMessageReceiver is IMultiMessageReceiver, ExecutorAware, Initializ
...
-     * @notice Update bridge receiver adapters.
-     * This function can only be called by executeMessage() invoked within receiveMessage() of this contract,
-     * which means the only party who can make these updates is the caller of the MultiMessageSender at the source chain.
-     */
-    function updateReceiverAdapter(
-        address[] calldata _receiverAdapters,
-        bool[] calldata _operations
-    ) external onlySelf {
+    /// @notice Update bridge receiver adapters.
+    /// @dev This function can only be called by executeMessage() invoked within receiveMessage() of this contract,
```
Comment (ermyas) [inline src/MultiMessageReceiver.sol:138]: > executeMessage() invoked within receiveMessage()

`executeMessage()` is not called within `receiveMessage()` any longer, so can we update the comment accordingly? There are a few other areas where a similar comment exists, which we might also want to update.

**Review by ermyas — APPROVED** [review]
LGTM overall. Added a minor comment that we might want to address before merging. Also, it would be good to add the `out/` directory to `.gitignore` and remove it from this PR.

**Comment by 0xkydo** [comment]
Other than the things @ermyas mentioned. Just two things:
- In the README file, compiling should be `forge build` instead of `forge compile`
- The testing scripts should be under folder `test`, not `script`. Unless we want to change the toml config.

**Comment by 0xkydo** [comment]
lgtm

**Review by 0xkydo — APPROVED** [review]

**Pull request merged**